tests/infra/test_with_custom_state.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+
+from eth2spec.test.context import (
+    with_custom_state,
+    default_activation_threshold,
+    default_balances,
+    zero_activation_threshold,
+    _custom_state_cache_dict,
+)
+from eth2spec.test.helpers.constants import MINIMAL, PHASE0
+from eth2spec.test.helpers.specs import spec_targets
+
+
+@pytest.fixture(autouse=True)
```

Comment:
I am afraid that autouse will make it run for every test, not just the ones in this file. Please use the setup tear down in test classes like in another exmples